CMakeLists.txt

```diff
@@ -117,6 +117,11 @@ if( USE_CUDA )
     list( APPEND HIP_INCLUDE_DIRS "${HIP_ROOT_DIR}/include" )
 endif( )
 
+# HIP on Windows: xhip is required with clang++ to get __half defined
+if ( WIN32 )
+  add_compile_options(-xhip)
+endif()
```

Comment:
These blanket options are normally discouraged.   Can you set on targets if this is required?